tools/update_runnablejobs.py

```diff
@@ -1,98 +1,201 @@
+        key = _unique_key(new_job)
+
+        # Let's build a map of all the jobs and if duplicated change the build_system_type to *
+        if key not in map:
+            map[key] = job['build_system_type']
+            sanitized_list.append(new_job)
+
+        elif job['build_system_type'] != map[key]:
+            previous_job = copy.deepcopy(new_job)
+            previous_job['build_system_type'] = map[key]
+            previous_index = sanitized_list.index(previous_job)
+            # This will make so we *replace* the previous build system type with '*'
+            # This also guarantees that we don't have duplicates
+            sanitized_list[previous_index]['build_system_type'] = '*'
+
+    return sanitized_list
+
+
+def query_sanitized_data(repo_name='mozilla-inbound'):
+    '''Return sanitized jobs data based on runnable api. None if failed to obtain or no new data.
+
+     We need to find the latest gecko decision task ID (by querying the index [1][2])
+     in order to know which task ID to pass to the runnable api [3][4].
+
+     It stores the minimal sanitized data from runnable apis under ~/.mozilla/seta/<task_id>.json
+
+     [1] https://index.taskcluster.net/v1/task/gecko.v2.%s.latest.firefox.decision/
+     [2] Index's data structure:
+      {
+        "namespace": "gecko.v2.mozilla-inbound.latest.firefox.decision",
+        "taskId": "Dh9ZvFk5QCSprJ877cgUmw",
+        "rank": 0,
+        "data": {},
+        "expires": "2017-10-06T18:30:18.428Z"
+      }
+     [3] https://treeherder.mozilla.org/api/project/mozilla-inbound/runnable_jobs/?decision_task_id=Pp7ZxoH0SKyU6wnhX_Fp0g&format=json
+     [4] NOTE: I Skip some data that is not relevant to failures.py
+     {
+      meta: {
+        count: 2317,
+        offset: 0,
+        repository: mozilla-inbound,
+      },
+      results: [
+        {
+          build_platform: windows8-64,
+          build_system_type: buildbot,
+          job_type_name: W3C Web Platform Tests,
+          platform_option: debug,
+          ref_data_name: Windows 8 64-bit mozilla-inbound debug test web-platform-tests-1,
+        },
+        {
+            "build_platform": "linux64",
+            "build_system_type": "taskcluster",
+            "job_type_name": "desktop-test-linux64/opt-reftest-8",
+            "platform": "linux64",
+            "platform_option": "opt",
+            "ref_data_name": "desktop-test-linux64/opt-reftest-8",
+        },
+    '''
     url = "https://index.taskcluster.net/v1/task/gecko.v2.%s.latest.firefox.decision/"
     try:
-        latest_task = retry(requests.get, args=(url % "mozilla-inbound", ),
-                            kwargs={'headers': {'accept-encoding': 'json'},
-                                    'verify': True}).json()
+        latest_task = retry(
+            requests.get,
+            args=(url % repo_name, ),
+            kwargs={'headers': {'accept-encoding': 'json'}, 'verify': True}
+        ).json()
+        task_id = latest_task['taskId']
```

Comment:
We were also doing the same previously.
We were checking for a timestamp which was being updated every time there is a new task.
The latest index always points to the latest gecko decision task.

What we could do is track if the size of the response has changed from the last time.

In any case, once we're on SETA we can probably query directly.